### DISCUSSION TITLE
Support reverting  commied changes.

### DISCUSSION THREAD

**Discussion by ghost** [magit/magit#4497] [Support] [answered]
I would like to ask for a very simple thing:

Suppose, you found out that there is a regression, and a file is broken.

You go to a file in dired, type `magit-dispatch`, type `l -- filename l` to find exactly the commit that caused that regression. Now you want to revert that changes. The most natural thing is to type `k` to "discard" that change, however, magit says:

```
user-error: Cannot discard committed changes
```

I would like magit to revert this file to the previous (last non-broken) revision.
Is that possible?

**Comment by tarsius** [comment, marked as answer]
It cannot be discarded yes, but it can of course already be reversed. The key is `v`. (Or revert the commit as such using `V`.)

The various apply variants are discussed at https://magit.vc/manual/magit/Applying.html.
